### PR TITLE
chore(store): add typing for the Action decorator

### PR DESCRIPTION
**Pull request by deebloo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ x] Other... Please describe:
```
Add updated types for Action

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
The Action decorator doesn't accept "any" anymore it accepts and ActionDef

## Does this PR introduce a breaking change?
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: packages/store/src/symbols.ts [hunk 1]
```diff
@@ -44,6 +44,12 @@ export class NgxsConfig {
   }
 }
 
+export interface ActionDef {
```
Comment (arturovt) [inline packages/store/src/symbols.ts]: you could also do `export type ActionDef<T = any> = Type<T> & { type: string };`

Comment (deebloo) [reply packages/store/src/symbols.ts]: true we could

Comment (arturovt) [reply packages/store/src/symbols.ts]: @deebloo `store/src/store.ts` uses `Type` from Angular lib, but it's up to you :wink:

Comment (eranshmil) [reply packages/store/src/symbols.ts]: I think we should decrease the angular dependency, in order to someday make the library available also for other libraries/frameworks.

Comment (splincode) [reply packages/store/src/symbols.ts]: @eranshmil hardly

Comment (eranshmil) [reply packages/store/src/symbols.ts]: All the angular stuff should be removed from the core and moved to a plugin.

**Comment by arturovt** [comment]
@eranshmil I understand what you're talking about, but the DI in this library works only thanks to Angular

**Comment by eranshmil** [comment]
> @eranshmil I understand what you're talking about, but the DI in this library works only thanks to Angular

Are you talking about the DI in the applications that use NGXS?
The DI is not relevant in React application, for instance.

**Comment by arturovt** [comment]
@eranshmil I'm talking about the DI in the NGXS library

**Pull request merged**